flink-runtime/src/main/java/org/apache/flink/runtime/state/memory/MemoryBackendCheckpointStorage.java

```diff
@@ -117,6 +122,11 @@ public CheckpointStorageLocation initializeLocationForCheckpoint(long checkpoint
 			// prepare all the paths needed for the checkpoints
 			checkState(fileSystem != null);
 
+			if (!isDirectoryCreated) {
+				fileSystem.mkdirs(checkpointsDirectory);
```

Comment:
My comments/question about lazy initialization also apply to this class.

Reply:
Comments about some unnecessary IOExceptions apply to some methods in the class as well.